### PR TITLE
Metrics

### PR DESCRIPTION
**Pull request by azfarkhoja305**
Metrics: Added inception score + faster FID calculation.

Moved metric calculation scripts from utils/ to metrics/

Model layers.py from utils/ to models/

Added a seed function in utils/utils.py

Changes in datasets,py, regarding train and valid transforms

Created a notebook (Metrics.ipynb) to show the changes.

### CODE REVIEW

File: metrics/torch_inception.py [hunk 1]
```diff
@@ -72,15 +75,16 @@ def __init__(self,
...
+        #     'Last possible output block index is 3'
 
         self.blocks = nn.ModuleList()
 
         if use_fid_inception:
             inception = fid_inception_v3()
         else:
-            inception = models.inception_v3(pretrained=True)
+            inception = models.inception_v3(
+                pretrained=True, init_weights=False)
```
Comment (notAlex2) [inline metrics/torch_inception.py:87]: Why set init_weights=False?

File: metrics/torch_inception.py [hunk 2]
```diff
@@ -24,10 +26,11 @@ class InceptionV3(nn.Module):
...
     BLOCK_INDEX_BY_DIM = {
-        64: 0,  # First max pooling features
-        192: 1,  # Second max pooling featurs
-        768: 2,  # Pre-aux classifier features
-        2048: 3  # Final average pooling features
+        64: 0,      # First max pooling features
+        192: 1,     # Second max pooling featurs
+        768: 2,     # Pre-aux classifier features
+        2048: 3,    # Final average pooling features
+        'prob': 4,  # softmax layer
```
Comment (notAlex2) [inline metrics/torch_inception.py:33]: Why was softmax added and why wasn't it there before?

File: metrics/torch_inception.py [hunk 3]
```diff
@@ -159,6 +167,15 @@ def forward(self, inp):
             if idx == self.last_needed_block:
                 break
 
+        if self.last_needed_block >= 4:
+            x = F.dropout(x, training=self.training)
```
Comment (notAlex2) [inline metrics/torch_inception.py:171]: Dropout for FID? Will we ever train the model?

File: metrics/torch_is_fid_score.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
+# Modified from
```
Comment (notAlex2) [inline metrics/torch_is_fid_score.py:1]: Is this file related to utils/torch_fid_score.py?
